lib/connect.serv.js

```diff
@@ -0,0 +1,299 @@
+          return thenDo[action]();
+        } else {
+          // unknown error here
+        }
+      }
+    });
+
+    return {
+      providerName: "MM",
+      connectedAddress: requestAcc,
+      connectedChain: getChain.connectedChain,
+      chainId: getChain.chainId,
+      providerInstance: providerInstance,
+      status: 'connected'
+    }
+    // Else is Wallet-Connect
+  } else {
+    const wcProviderNext = await providerInstance.currentProvider.enable().then((res) => {
+      // TEMP! For testing WC connection  
+      // if (claimAddress !== res[0]){
+      //   return {
+      //     providerName: "WC",
+      //     connectedAddress: res[0],
+      //     connectedChain: chainConnected,
+      //     chainId: providerInstance.chainId,
+      //     providerInstance: providerInstance
+      //   }
+      // }
+      if (claimAddress !== res[0]) {
+        providerName = "WC";
+        wrongAddressError(providerInstance, providerName);
+      } 
+      else {
+        let chainConnected = providerInstance.currentProvider.chainId === 1 ? "Ethereum Mainnet" : "Fuse";
+        return {
+          providerName: "WC",
+          connectedAddress: res[0],
+          connectedChain: chainConnected,
+          chainId: providerInstance.currentProvider.chainId,
+          providerInstance: providerInstance,
+          status: 'connected'
+        }
+      }
+    });
+    return wcProviderNext;
+  }
+}
+
+/**
+ * @notice Checks if a user has an existing connection
+ * @param claimAddress address with GOOD tokens to claim
+ * @return provider instance and connect wallet information
+ */
+
+export async function isConnected(claimAddress) {
+  if (claimAddress !== null) {
+    let providerName, providerInstance, tryMetamask;
+    const web3 = new Web3(Web3.givenProvider || infuraConfig.infuraUrl);
+    const Wc3 = new WalletConnectProvider({
+      infuraId: infuraConfig.infuraId,
+      rpc: {
+        1: infuraConfig.infuraUrl,
+        122: infuraConfig.pocketUrl,
+      }
+    });
+    const web3wc = new Web3(Wc3);
+
+    let tryWalletConnect = Wc3.wc.accounts;
+    if (!tryWalletConnect[0]) {
+      tryMetamask = await web3.eth.getAccounts();
+    }    
+    const isConnected = tryWalletConnect[0] ? 'WC' : tryMetamask[0] ? 'MM' : null;
+    if (isConnected) {
+      providerName = isConnected,
+      providerInstance = tryWalletConnect[0] ? web3wc : web3;
+      const providerInit = walletConnect(providerName, providerInstance, claimAddress);
+      return providerInit;
+    }
+  }
+}
+
+/**
+ * @notice Checks if a user already claimed their tokens on a network
+ * @param currentConnection contains all the necessary information of the connected wallet
+ * @return boolean's for claim-status per network
+ */
+export async function getClaimStatus(currentConnection) {
+  let contractAddressess = {
+        fuse: '0x3A9299BE789ac3730e4E4c49d6d2Ad1b8BC34DFf',
+        rootState: '0x3A9299BE789ac3730e4E4c49d6d2Ad1b8BC34DFf'
```

Comment:
is this the contract address? why is it hardcoded? why is it called "rootState"?
the addresses should be read from @gooddollar/goodprotocol/releases/deploymnet.json
the keys in that json are:
staging - fuse staging contracts
staging-mainnet staging contracts on ropsten that simulate mainnet
production - fuse production contracts
production-mainnet ethereum production contracts

Reply:
confused contract names and name of hash.
changed now for clarity